src/vs/code/node/cli.ts

```diff
@@ -319,23 +324,61 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
-				}
-			});
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = verbose || hasReadStdinArg;
+			const openArgs: string[] = ['-n'];
+
+			let tmpStdoutLogger: CliVerboseLogger | undefined;
+			let tmpStderrLogger: CliVerboseLogger | undefined;
+			if (requiresWait) {
+				const tmpStdoutName = createFileName(tmpdir(), 'code-stdout');
+				const tmpStderrName = createFileName(tmpdir(), 'code-stderr');
+				tmpStdoutLogger = new CliVerboseLogger(tmpStdoutName);
+				tmpStderrLogger = new CliVerboseLogger(tmpStderrName);
+				openArgs.push('-W');
+				openArgs.push('--stdout', tmpStdoutName);
+				openArgs.push('--stderr', tmpStderrName);
+			}
+			const argsArr: string[] = [];
+			const isDev = env['NODE_ENV'] === 'development';
```

Comment:
Shouldn't this be `VSCODE_DEV` ?

Reply:
`code-cli.sh` does both. It sets `VSCODE_DEV` to 1 and `NODE_ENV` to `development`.
I guess `VSCODE_DEV` is preferable because it's more specific

Reply:
Yes, rely on vscode specifc variables